docs/machine-learning/how-to-guides/inspect-intermediate-data-ml-net.md

```diff
@@ -0,0 +1,84 @@
+---
+title: Inspect intermediate data values during ML.NET pipeline processing
+description: Learn how to inspect actual intermediate data values during ML.NET machine learning pipeline processing
+ms.date: 11/07/2018
+ms.topic: how-to
+ms.custom: mvc
+#Customer intent: As a developer, I want to inspect actual intermediate data values during ML.NET machine learning pipeline processing so that I can make sure that I'm getting the results I expect.
+---
+# Inspect intermediate data values during ML.NET pipeline processing
+
+During the experiment, you may want to observe and validate the data processing results at a given point. This isn't easy since ML.NET operations are lazy, constructing objects that are 'promises' of data.
+
+The `GetColumn<T>` extension method lets you inspect the intermediate data. It returns the contents of one data column as an `IEnumerable`.
+
+The following example shows how to use the `GetColumn<T>` extension method:
+
+[Example file](../../test/data/adult.tiny.with-schema.txt):
+```
+Label	Workclass	education	marital-status
+0	Private	11th	Never-married
+0	Private	HS-grad	Married-civ-spouse
+1	Local-gov	Assoc-acdm	Married-civ-spouse
+1	Private	Some-college	Married-civ-spouse
+
+```
+
+Our class is defined as follows:
+
+```csharp
```

Comment:
These larger samples should be extracted from working code in the [samples](https://github.com/dotnet/samples) repo.

That way, they go through the CI builds for validation, and customers can download the working samples.